core/src/main/scala/org/apache/spark/sql/delta/files/TahoeCDCAddFileIndex.scala

```diff
@@ -16,30 +16,34 @@
 
 package org.apache.spark.sql.delta.files
 
-import org.apache.spark.sql.delta.{DeltaLog, Snapshot}
-import org.apache.spark.sql.delta.actions.AddFile
-import org.apache.spark.sql.delta.commands.cdc.CDCReader
-import org.apache.spark.sql.delta.commands.cdc.CDCReader._
-import org.apache.spark.sql.delta.implicits._
 import org.apache.hadoop.fs.Path
 
 import org.apache.spark.sql.SparkSession
 import org.apache.spark.sql.catalyst.expressions.Expression
+import org.apache.spark.sql.delta.actions.AddFile
+import org.apache.spark.sql.delta.commands.cdc.CDCReader
+import org.apache.spark.sql.delta.commands.cdc.CDCReader._
+import org.apache.spark.sql.delta.implicits._
+import org.apache.spark.sql.delta.{DeltaLog, Snapshot}
 import org.apache.spark.sql.types.StructType
 
 /**
  * A [[TahoeFileIndex]] for scanning a sequence of added files as CDC. Similar to
  * [[TahoeBatchFileIndex]], with a bit of special handling to attach the log version
  * and CDC type on a per-file basis.
  */
-class CdcAddFileIndex(
+class TahoeCDCAddFileIndex(
     override val spark: SparkSession,
     filesByVersion: Seq[CDCDataSpec[AddFile]],
     deltaLog: DeltaLog,
     path: Path,
     snapshot: Snapshot)
-  extends TahoeBatchFileIndex(
```

Comment:
I think this is major issue. `CdcAddFileIndex` extends `TahoeBatchFileIndex` while the others extend `TahoeFileIndex`. However, after refactoring, all of them extend `TahoeFileIndex`. I am still trying to look at code of both of those classes to understand whats the difference this is creating. Until we have a clear handle on that, we cant combine CdcAddFileIndex under the same base class. Can you dig into this and understand the differences?

Reply:
@tdas I couldn't find (or understand) the difference between these two classes. I must admit that I'm also new to the code base.

Reply:
Hi @amirmor1 - this is likely a blocker of this PR then. I'm hesitant to merge a change where we are unsure of the semantic/structural implications of the change. WDYT?